audit/options.go

```diff
@@ -158,3 +191,161 @@ func WithHeaderFormatter(f HeaderFormatter) Option {
 		return nil
 	}
 }
+
+// WithExclusions provides an Option to supply exclusions in a JSON string format.
+// See 'exclusion' type for more information and example below:
+// Expected JSON format:
+//
+//	[
+//		{
+//			"condition": "\"/request/mount_type\" == transit",
+//			"fields": [ "/request/data", "/response/data" ]
+//		},
+//		{
+//			"condition":  "\"/request/mount_type\" == userpass",
+//			"fields": [ "/request/data" ]
+//		}
+//	]
+func WithExclusions(e string) Option {
+	return func(o *options) error {
+		e = strings.TrimSpace(e)
+		if e == "" {
+			return nil
+		}
+
+		var result []*exclusion
+
+		err := json.Unmarshal([]byte(e), &result)
+		if err != nil {
+			return fmt.Errorf("unable to parse exclusions: %w", err)
+		}
+
+		// Validate the exclusions
+		for _, exc := range result {
+			if err := exc.validate(); err != nil {
+				return err
+			}
+		}
+
+		o.withExclusions = result
+
+		return nil
+	}
+}
+
+// UnmarshalJSON handles unmarshalling JSON bytes (string representation) of a collection
+// of exclusion types into a Go type.
+func (e *exclusion) UnmarshalJSON(b []byte) error {
+	// Reference the JSON struct tags for exclusion.
+	const keyFields = "fields"
+	const keyCondition = "condition"
+
+	var err error
+
+	m := make(map[string]any)
+	if err = json.Unmarshal(b, &m); err != nil {
+		return err
+	}
+
+	// Parse fields
+	f, ok := m[keyFields]
+	if !ok {
+		return fmt.Errorf("exclusion '%s' missing", keyFields)
+	}
+	intermediateFields, ok := f.([]any)
+	if !ok {
+		return fmt.Errorf("unable to parse '%s': expected collection of %s; got: '%v'", keyFields, keyFields, f)
```

Comment:
Nit/PSA: you can use `%q` instead of `'%s'` (applies above too)

Reply:
👍🏼 yeah, that's a good one! the only reason I'm hanging onto the `'%s` at the moment is that there were a few places where we output data that has escaped `"` and I just thought it looked a bit much. This example you pointed out isn't one of those however... not sure if it's nicer to keep them all single quotes in this area or not 🤔

Reply:
Yeah that makes sense! Mostly pointing it out since I always forget about `%q` so making sure it was in our collective minds :P